internal/installercache/installercache.go

```diff
@@ -232,3 +327,42 @@ func (i *Installers) evictFile(filePath string) error {
 	}
 	return nil
 }
+
+// pruneExpiredHardLinks removes any hardlinks that have been around for too long
+// the grace period is used to determine which links should be removed.
+func (i *Installers) pruneExpiredHardLinks(links []*fileInfo, gracePeriod time.Duration) {
+	for idx := 0; idx < len(links); idx++ {
+		finfo := links[idx]
+		graceTime := time.Now().Add(-1 * gracePeriod)
+		grace := graceTime.Unix()
+		if finfo.info.ModTime().Unix() < grace {
+			os.Remove(finfo.path)
```

Comment:
I know we were not checking this before but should we be checking this error?
Also if we think we really shouldn't have to ever run this should we maybe log if we do end up deleting something here?

Reply:
We can certainly log something but you should also know that I plan to record this in the metrics ticket.